datasets/nz/designated_terrorists/crawler.py

```diff
@@ -0,0 +1,91 @@
+    table: _Element,
+) -> Generator[Dict[str, Tuple[str, Optional[str]]], None, None]:
+    headers = None
+    for row in table.findall(".//tr"):
+        if headers is None:
+            headers = []
+            for el in row.findall("./th"):
+                headers.append(slugify(el.text_content()))
+            continue
+
+        cells = []
+        for el in row.findall("./td"):
+            for span in el.findall(".//span"):
+                # add newline to split spans later if we want
+                span.tail = "\n" + span.tail if span.tail else "\n"
+
+            # there can be multiple links in the same cell
+            a_tags = el.findall(".//a")
+            if a_tags is None:
+                cells.append((el.text_content(), None))
+            else:
+                cells.append((el.text_content(), [a.get("href") for a in a_tags]))
+
+        assert len(headers) == len(cells)
+        yield {hdr: c for hdr, c in zip(headers, cells) if hdr}
+
+
+def crawl_item(input_dict: dict, context: Context):
+    # aliases will be either a list of size one or None if there is no aliases
+    name, *aliases = (
+        input_dict.pop("terrorist-entity")[0].strip().split("\n\t\t\t\t", 1)
+    )
+
+    # We remove the prefix and split in the commas
+    aliases = aliases[0].replace("Also known as ", "").split(", ") if aliases else []
+
+    organization = context.make("Organization")
+    organization.id = context.make_slug(name)
+    organization.add("topics", "crime")
+
+    organization.add("name", name)
+    for alias in aliases:
+        organization.add("alias", alias)
+
+    sanction = h.make_sanction(context, organization)
+
+    raw_initial_sanction_date, initial_statement_url = input_dict.pop(
+        "date-of-designation-as-a-terrorist-entity-in-new-zealand-including-and-statement-of-case-for-designation"
+    )
+
+    initial_sanction_date = re.findall(DATE_PATTERN, raw_initial_sanction_date)[0]
+
+    # There is only one date in this case
+    sanction.add("startDate", h.parse_date(initial_sanction_date, formats=["%d %B %Y"]))
+    sanction.add("sourceUrl", initial_statement_url)
+
+    raw_renew_sanction_dates, renew_statement_urls = input_dict.pop(
+        "date-terrorist-designation-was-renewed-in-new-zealand-including-statement-of-case-for-renewal-of-designation"
+    )
+
+    renew_sanction_dates = re.findall(DATE_PATTERN, raw_renew_sanction_dates)
+
+    for renew_sanction_date in renew_sanction_dates:
+        sanction.add("date", h.parse_date(renew_sanction_date, formats=["%d %B %Y"]))
+
+    for renew_statement_url in renew_statement_urls:
+        sanction.add("sourceUrl", renew_statement_url)
+
+    context.emit(organization, target=True)
+    context.emit(sanction)
+    context.audit_data(input_dict)
+
+
+def crawl(context: Context):
+    response = context.fetch_html(context.data_url)
+
+    response.make_links_absolute(context.data_url)
+
+    for item in parse_table(response.find(".//table")):
```

Comment:
could you add an assertion that the table contains a caption with the text "Alphabetical list of Designated Terrorist Entities in New Zealand pursuant to UNSC Resolution 1373" to be 100% sure we use the correct table? Just in case something on the page changes and we accidentally start emitting the table of people who have been delisted as terrorists.